cf-custom-resources/lib/dns-delegation.js

```diff
@@ -0,0 +1,250 @@
+'use strict';
+
+const aws = require('aws-sdk');
+
+const defaultSleep = function (ms) {
+    return new Promise(resolve => setTimeout(resolve, ms));
+};
+
+// These are used for test purposes only
+let defaultResponseURL;
+let waiter;
+
+
+/**
+ * Upload a CloudFormation response object to S3.
+ *
+ * @param {object} event the Lambda event payload received by the handler function
+ * @param {object} context the Lambda context received by the handler function
+ * @param {string} responseStatus the response status, either 'SUCCESS' or 'FAILED'
+ * @param {string} physicalResourceId CloudFormation physical resource ID
+ * @param {object} [responseData] arbitrary response data object
+ * @param {string} [reason] reason for failure, if any, to convey to the user
+ * @returns {Promise} Promise that is resolved on success, or rejected on connection error or HTTP error response
+ */
+let report = function (event, context, responseStatus, physicalResourceId, responseData, reason) {
```

Comment:
How much of a pain would it be to move this to a separate file that's shared between the two lambdas?

Reply:
I think it'd be pretty tough - it'd also make the packaging step a bit more complicated.